compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -306,11 +306,56 @@ pub fn from_fn_attrs<'ll, 'tcx>(
         // Need this for AArch64.
         to_add.push(llvm::CreateAttrStringValue(cx.llcx, "branch-target-enforcement", "false"));
     }
-    if codegen_fn_attrs.flags.contains(CodegenFnAttrFlags::ALLOCATOR) {
+    if codegen_fn_attrs.flags.contains(CodegenFnAttrFlags::ALLOCATOR)
+        || codegen_fn_attrs.flags.contains(CodegenFnAttrFlags::ALLOCATOR_ZEROED)
+    {
+        if llvm_util::get_version() >= (15, 0, 0) {
+            to_add.push(llvm::CreateAttrStringValue(cx.llcx, "alloc-family", "__rust_alloc"));
```

Comment:
I'd suggest extracting a small `create_alloc_family_attr()` function, so we need to hardcode the `__rust_alloc` family only once.